sov-modules/sov-modules-macros/src/dispatch/derive_rpc.rs

```diff
@@ -0,0 +1,285 @@
+    let intermediate_trait_name = format_ident!("{}Rpc", type_name);
+
+    let wrapped_attr_args = quote! {
+        (#attrs)
+    };
+    let rpc_attribute = syn::Attribute {
+        pound_token: syn::token::Pound { spans: [proc_macro2::Span::call_site()] },
+        style: syn::AttrStyle::Outer,
+        bracket_token: syn::token::Bracket { span: proc_macro2::Span::call_site() },
+        path: jsonrpsee_rpc_macro_path(),
+        tokens: wrapped_attr_args,
+    };
+    // Iterate over the methods from the `impl` block, building up three lists of items as we go
+
+    let generics = &input.generics;
+    let mut rpc_info = RpcImplBlock {
+        type_name,
+        methods: vec![],
+        working_set_type: None,
+        generics: generics.clone(),
+    };
+
+    let mut intermediate_trait_items = vec![];
+    let mut simplified_impl_items = vec![];
+    for item in input.items.into_iter() {
+        if let ImplItem::Method(ref method) = item {
+            if let Some((attr, idx_of_rpc_attr)) = get_method_attribute(&method.attrs) {
+                let idx_of_working_set_arg = find_working_set_argument(&method.sig);
+                if let Some(idx) = idx_of_working_set_arg {
+                    let arg = method.sig.inputs.iter().skip(idx).next().expect("WorkingSet arg was just verified to be present");
+                    if let FnArg::Typed(PatType { ty, .. }) = arg {
+                       rpc_info.working_set_type = Some(*ty.clone());
+                    }
+                }
+                rpc_info.methods.push(RpcEnabledMethod{
+                    method_name: method.sig.ident.clone(),
+                    method_signature: method.sig.clone(),
+                    idx_of_working_set_arg
+                });
+                let mut intermediate_signature = method.sig.clone();
+                
+                let annotated_signature = quote! {
+                    #attr
+                    #intermediate_signature;
+                };
+                intermediate_trait_items.push(annotated_signature);
+                
+                let mut original_method = method.clone();
+                original_method.attrs.remove(idx_of_rpc_attr);
+                simplified_impl_items.push(ImplItem::Method(original_method));
+                continue
+            } 
+        }
+        simplified_impl_items.push(item)
+    }
+
+    let impl_rpc_trait_impl = rpc_info.build_rpc_impl_trait();
+
+    // Replace the original impl block with a new version with the rpc_gen and related annotations removed
+    input.items = simplified_impl_items;
+    let simplified_impl = quote! {
+        #input
+    };
+   
+    let rpc_output = quote! {
+        #simplified_impl
+
+        #impl_rpc_trait_impl
+
+        #rpc_attribute
+        pub trait #intermediate_trait_name  #generics {
+
+            #(#intermediate_trait_items)*
+
+            #[method(name = "health")]
+            fn health(&self) -> ::jsonrpsee::core::RpcResult<()> {
```

Comment:
What do you think about checking connection to the database in health call?

Reply:
Since we use RocksDB, there isn't really the same notion of a connection - since it's embedded, you can't disconnect without having massive disk corruption on the machine running the RPC server

Reply:
Oh, that makes sense.